Automattic/wp-calypso
# Checkout: Convert getThankYouUrl to TypeScript

#### Changes proposed in this Pull Request

In this PR we convert `useGetThankYouUrl` and `getThankYouPageUrl` (which the hook uses) to TypeScript. The logic behind these functions is pretty complex, so this will help to prevent bugs in the future.

This was extracted from https://github.com/Automattic/wp-calypso/pull/46375

#### Testing instructions

This should not have any user-facing changes. 

- Make sure the tests pass for the `getThankYouPageUrl` function.
  - `yarn run test-client client/my-sites/checkout/composite-checkout/test/composite-checkout-thank-you.js`
- Visit checkout and complete a purchase. Make sure you are redirected to a thank-you page of some sort.
